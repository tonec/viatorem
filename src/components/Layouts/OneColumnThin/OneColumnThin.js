import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import classNames from 'classnames'
import { Layout, Row, Col } from 'antd'
import { Navigation } from 'containers'

import styles from './styles.scss'

const { Header, Content } = Layout

const propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

const defaultProps = {
  className: ''
}

const OneColumnThin = ({ title, className, children, ...props }) => {
  const classes = classNames('one-column-thin', className)

  return (
    <div className={classes} {...props}>
      <Helmet title={title} />
      <Header className={styles.header}>
        <Navigation />
      </Header>
      <Content>
        <Row>
          <Col span={12} offset={6}>
            {children}
          </Col>
        </Row>
      </Content>
    </div>
  )
}

OneColumnThin.propTypes = propTypes
OneColumnThin.defaultProps = defaultProps

export default OneColumnThin
