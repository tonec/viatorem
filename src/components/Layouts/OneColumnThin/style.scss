.header {
  height: 46px;
}

.content {
  padding-top: 20px;
}