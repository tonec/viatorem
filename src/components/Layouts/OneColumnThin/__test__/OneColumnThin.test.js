import React from 'react'
import { shallow } from 'enzyme'
import Helmet from 'react-helmet'
import OneColumnThin from '../OneColumnThin'

const props = {
  title: 'Test title',
  className: 'test-class-name'
}

const childContent = <p>Child content</p>

describe('OneColumnThin (snapshot)', () => {
  it('renders according to snapshot', () => {
    const component = shallow(<OneColumnThin {...props}>{childContent}</OneColumnThin>)
    expect(component).toMatchSnapshot()
  })
})

describe('OneColumnThin (rendering)', () => {
  let component

  beforeEach(() => {
    component = shallow(<OneColumnThin {...props}>{childContent}</OneColumnThin>)
  })

  it('renders the correct title', () => {
    expect(component.find(Helmet)).toHaveLength(1)
    expect(component.find(Helmet).props().title).toBe(props.title)
  })

  it('includes the correct class name', () => {
    expect(component.find('.one-column-thin')).toHaveLength(1)
  })
})
