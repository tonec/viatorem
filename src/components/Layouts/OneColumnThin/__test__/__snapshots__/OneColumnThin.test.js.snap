// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneColumnThin (snapshot) renders according to snapshot when enabled 1`] = `
<div
  className="one-column-thin test-class-name"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Test title"
  />
  <Adapter>
    <Connect(Navigation) />
  </Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={6}
        span={12}
      >
        <p>
          Child content
        </p>
      </Col>
    </Row>
  </Adapter>
</div>
`;
